### PR TITLE
Docs and internal API cleanup: context/HTTP improvements, cookie overhaul, ESLint guard

### PR DESCRIPTION
**Pull request by bufferings**
Summary

- Improve context and HTTP APIs documentation and internal structure
- Add ESLint rule to prevent leaking internal APIs from barrels
- Overhaul cookie utilities with RFC 6265 compliance and Result-based errors
- Pin @changesets/cli to exact version

Details

Core (@korix/kori)
- Add comprehensive TSDoc to KoriRequest, KoriResponse, and context modules with examples
- Apply @packageInternal tags to framework infrastructure functions
- Remove internal factory functions from public exports (createKoriEnvironment, createKoriHandlerContext, createKoriInstanceContext, createKoriRequest, createKoriResponse)
- HTTP utilities:
  - Replace method helper with getMethodString in packages/kori/src/http/methods.ts
  - Remove unused accept.ts and legacy method.ts
  - Cleanup status-codes and headers utilities
- Request/Response improvements:
  - Response builders set default content-type per body kind when not explicitly set
  - Error helpers build structured JSON bodies
  - Request helpers provide cached accessors for url, method, params, headers, cookies, and typed body parsing

Cookies (@korix/kori)
- RFC 6265 compliant parsing and serialization
- Result-based error handling with structured error types
- Type-safe prefix constraints __Secure- and __Host-
- 400-day limit enforcement for Max-Age and Expires
- Add deleteCookie and strengthen setCookie validation
- Update tests to cover edge cases

ESLint config (@korix/eslint-config)
- Add new no-barrel-internal rule to prevent exporting symbols marked @internal or @packageInternal from barrels
- Integrate the rule into kori config
- Add TSDoc guideline file (.cursor/rules/tsdoc.mdc)

Workspace
- Pin @changesets/cli from ^2.29.5 to 2.29.5 and update lockfile

Tests
- Update cookies tests and logging lazy-initialization test
- Adjust file-plugin-nodejs tests

Affected files (34)
- Adds: .changeset/add-no-barrel-internal-rule.md, .changeset/improve-context-api-docs.md, .cursor/rules/tsdoc.mdc, packages/kori/src/http/methods.ts
- Removes: packages/kori/src/http/accept.ts, packages/kori/src/http/method.ts
- Modifies: many files under packages/kori/src/context, packages/kori/src/http, packages/eslint-config, plus root package.json and pnpm-lock.yaml

Notes
- Lint, format, and typecheck hooks pass on commit

Updates

Cookies (@korix/kori)
- Enforce SameSite=None requires Secure at type level and runtime
- Refactor CookieOptions typing to a single definition with union constraints
- Update tests to cover validation and success cases

## Summary by CodeRabbit

* **New Features**
  * Introduced a robust, RFC 6265-compliant cookie API with structured error handling and type-level constraints.
  * Added multiple new error response methods for common HTTP statuses.
  * Implemented a new ESLint rule to prevent exporting internal symbols from barrel files.
  * Added Result-based cookie parsing and serialization for safer error handling.

* **Improvements**
  * Enhanced and clarified documentation across context APIs, HTTP types, hooks, and error handling.
  * Expanded and refined type definitions and JSDoc for request, response, and context objects.
  * Improved logger and deferred callback management in context abstractions.

* **Bug Fixes**
  * Updated tests to reflect new cookie API and error handling behavior.
  * Corrected plugin names and response body assertions in file plugin tests.

* **Refactor**
  * Removed deprecated or unnecessary exports and constants.
  * Consolidated and clarified internal state and naming conventions.
  * Deleted unused HTTP utility files and migrated functions as needed.

* **Chores**
  * Updated dependency version pinning for development tools.
  * Removed unused package dependencies.

* **Documentation**
  * Added comprehensive guidelines for writing comments and JSDoc.
  * Improved API documentation with detailed examples and usage notes.

* **CI**
  * Added a new GitHub Actions workflow for continuous integration with build, lint, typecheck, and test steps.
  * Improved pnpm caching configuration in documentation deployment workflow.

### CODE REVIEW

File: packages/kori/src/http/cookies.ts [hunk 1]
```diff
@@ -1,187 +1,545 @@
...
-/**
- * Validates a cookie name according to RFC 6265.
- * Cookie names must contain only valid token characters.
- */
-function validateCookieName(name: string): void {
-  if (!name) {
-    throw new Error('Cookie name cannot be empty');
-  }
+      // Remove surrounding quotes if present
+      if (cookieValue.startsWith('"') && cookieValue.endsWith('"')) {
```
[the remark this answers is not in the document]
Comment (bufferings) [reply packages/kori/src/http/cookies.ts:245]: @Copilot hi
